src/tree.rs

```diff
@@ -139,6 +139,83 @@ impl<H: Hasher + Default, V: Value, S: Store<V>> SparseMerkleTree<H, V, S> {
         Ok(&self.root)
     }
 
+    /// Update multiple leaves at once
+    pub fn update_all(&mut self, mut leaves: Vec<(H256, V)>) -> Result<&H256> {
+        // Dedup(only keep the last of each key) and sort leaves
+        leaves.reverse();
+        leaves.dedup_by_key(|(a, _)| a.clone());
```

Comment:
I think this fn is being used incorrectly, according to [doc](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.dedup_by_key)
```
Removes all but the first of consecutive elements in the vector that resolve to the same key.

If the vector is sorted, this removes all duplicates.

Examples
let mut vec = vec![10, 20, 21, 30, 20];

vec.dedup_by_key(|i| *i / 10);

assert_eq!(vec, [10, 20, 30, 20]);
```

we should first call `sort_by_key`, then call `dedup_by_key`